src/features/loading/status-announcer.ts

```diff
@@ -0,0 +1,181 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ModelViewerElementBase from '../../model-viewer-base.js';
+import {debounce, getFirstMapKey} from '../../utils.js';
+
+export const INITIAL_STATUS_ANNOUNCEMENT =
+    'This page includes one or more 3D models that are loading';
+export const FINISHED_LOADING_ANNOUNCEMENT =
+    'All 3D models in the page have loaded';
+export const UPDATE_STATUS_DEBOUNCE_MS = 100;
+
+
+const $modelViewerStatusInstance = Symbol('modelViewerStatusInstance');
+const $updateStatus = Symbol('updateStatus');
+
+interface InstanceLoadingStatus {
+  instanceWasUnregistered: () => void;
+  loadAttemptCompletes: Promise<any>;
+}
+
+/**
+ * The LoadingStatusAnnouncer manages announcements of loading status across
+ * all <model-viewer> elements in the document at any given time. As new
+ * <model-viewer> elements are connected to the document, they are registered
+ * with a LoadingStatusAnnouncer singleton. As they are disconnected, the are
+ * also unregistered. Announcements are made to indicate the following
+ * conditions:
+ *
+ *  1. There are <model-viewer> elements that have yet to finish loading
+ *  2. All <model-viewer> elements in the page have finished attempting to load
+ */
+export class LoadingStatusAnnouncer {
+  /**
+   * The "status" instance is the <model-viewer> instance currently designated
+   * to announce the loading status of all <model-viewer> elements in the
+   * document at any given time. It might change as <model-viewer> elements are
+   * attached or detached over time.
+   */
+  protected[$modelViewerStatusInstance]: ModelViewerElementBase|null = null;
+
+  protected registeredInstanceStatuses:
+      Map<ModelViewerElementBase, InstanceLoadingStatus> =
+          new Map<ModelViewerElementBase, InstanceLoadingStatus>();
+
+  protected loadingPromises: Array<Promise<any>> = [];
+
+  /**
+   * This element is a node that floats around the document as the status
+   * instance changes (see above). It is a singleton that represents the loading
+   * status for all <model-viewer> elements currently in the page. It has its
+   * role attribute set to "status", which causes screen readers to announce
+   * any changes to its text content.
+   *
+   * @see https://www.w3.org/TR/wai-aria-1.1/#status
+   */
+  readonly statusElement: HTMLParagraphElement = document.createElement('p');
+  protected statusUpdateInProgress: boolean = false;
+
+  protected[$updateStatus]: () => void =
+      debounce(() => this.updateStatus(), UPDATE_STATUS_DEBOUNCE_MS);
+
+  constructor() {
+    this.statusElement.setAttribute('role', 'status');
```

Comment:
Do any browsers require any sort of element attributes before being used for ARIA props? (e.g. requires a minimum width/height or something)

Reply:
Visibility is important. The element will probably be ignored if you use `opacity: 0` or `visibility: hidden`, for example.